torchdynamo/_eval_frame.c

```diff
@@ -310,6 +286,23 @@ inline static PyObject *eval_custom_code(PyThreadState *tstate,
   return result;
 }
 
+static PyObject *_custom_eval_frame_shim(PyThreadState *tstate, PyFrameObject *frame,
+                                    int throw_flag) {
+// Shims logic into one of three states. Can probably be refactored into a single func, later:
+  //  - None: disables TorchDynamo
+  //  - False: run-only mode (reuse existing compiles)
+  //  - Python callable(): enables TorchDynamo
+    PyObject *callback = eval_frame_callback_get();
+    
+    if (callback == Py_None) {
+      return eval_frame_default(tstate, frame, throw_flag);
+    } else if (callback == Py_False) {
+      return _custom_eval_frame_run_only(tstate, frame, throw_flag);
+    } else {
+      return _custom_eval_frame(tstate, frame, throw_flag);
```

Comment:
You would add the arg here.  (Which lets you skip a TLS call inside the function.)

Reply:
Oh, I see, yes, I misunderstood you, good catch, we can skip the `eval_frame_callback_get()` downstream of the shim.